### ISSUE TITLE
Customize the json encoding of a self-defined struct in response

### ISSUE THREAD

**Issue by gossac** [swaggo/swag#845]
Hello, I'm new to swaggo and right now getting to know some amazing features of it. But I am trapped when trying to make my response structure displayed the same way MarshalJSON does. I've dived into documentations and issues for days, with no progress. Thanks in advance for any suggestions!

```go
type RespData struct {
    Idx int `resp:"index"`
}

type Resp struct {
    Code int `resp:"'code"`
    Data interface{}
    Msg string `resp:"-"`
}

type respPtr *Resp

func (resp *Resp) MarshalJSON() (jsonEncoding []byte, err error) {
    jsonConf := jsoniter.Config{
        EscapeHTML: true,
        SortMapKeys: true,
        ValidateJsonRawMessage: true,
        TagKey: "resp",
    }.Froze()
    jsonEncoding, err = jsonConf.Marshal((respPtr)(resp))
    return
}

// @Success 200 {object} Resp{data=RespData} "response data"
func (controller *Controller) HelloWorld(ctx *gin.Context) {}
```

The expected behavior of code above is to  encode fields into their "resp" tagged names(like a self-defined version of the "json" tag). It seems that MarshalJSON does not work in doc generation process. Any idea how to deal with this issue?

**Comment by ubogdan** [comment]
@gossac Swagger is not parsing any of the function body, you need to figure out a better way to do that. You can try to do this by embedding your generic response (the one with Msg and Code) in every route response.

**Comment by gossac** [comment]
@ubogdan Thanks for your reply. Actually the same problem would happen in `Data` field even if I embed `Resp`.

**Comment by gossac** [comment]
@ubogdan Thanks for your reply.

**Comment by gossac** [comment]
@ubogdan OK. Actually the same issue would happen recursively in `Data` field even if I embed `Resp`.

I think the underlying problem here is how to enable customized json tag for different routes. When I reuse a type, I embed it in multiple response structures and want it to be encoded in the corresponding way respectively(using different json tags). As for the example above, I may prevent some fields of `RespData` from shown when it is embeded in `Resp` but display them when `RespData` is placed in other responses.

Thank you for your help. I'll work on it further.
